external_services/lti.py

```diff
@@ -21,7 +25,33 @@ def __init__(self, service, user, instance, host, title, context_id=None, link_i
         link_id = link_id or "aplus{:d}".format(service.pk)
         title = title or link_id
 
-        student_id = self.external_student_id(user)
+        # Gather user information
+        student_id = "i" + self.external_student_id(user) # i for internal
+        full_name = "{} {}".format(user.first_name, user.last_name)
+        given_name = user.first_name
+        family_name = user.last_name
+        email = user.email
+
+        # Anonymize user information
+        if not service.allow_real_user_data:
+            enrollment = Enrollment.objects.filter(course_instance=instance, user_profile=user.userprofile).first()
+            if enrollment:
+                # Creates anon name and id for pre-anonymisation Enrollments
+                if not (enrollment.anon_name or enrollment.anon_id):
+                    # the model's post_save functions take care of the creation
+                    enrollment.save()
+                student_id = "a" + enrollment.anon_id # a for anonymous
+                full_name = enrollment.anon_name
+                names = enrollment.anon_name.split(" ")
+                given_name = names[0]
+                family_name = names[1] if len(names) > 1 else "Anonymous"
+                email = "{}-{}.{}@aplus.invalid".format(
```

Comment:
`anonymous-{anom_id}` should be enough. We can change the domain on a later date and then actually redirect emails to real users :)